tests/mock_code/test_diff.py

```diff
@@ -172,3 +186,59 @@ def test_regenerate_test_data(mock_code_factory, generate_diff_inputs, datadir):
     # check that ignore_paths works
     assert not (datadir / '_aiidasubmit.sh').is_file()
     assert (datadir / 'file1.txt').is_file()
+
+
```

Comment:
Instead of duplicating the `regenerate_test_data` code, one could use `@pytest.mark.parametrize` here.

I would guess something like
```python
@pytest.mark.parametrize("kwargs", [{'label': 'diff-relative'}, {...}])
def test_regenerate_test_data_relative(kwargs, mock_code_factory, generate_diff_inputs, datadir)
  mock_code = mock_code_factory(
      **kwargs,
      data_dir_abspath=...
```